### DISCUSSION TITLE
test nuclei reverse output no results

### DISCUSSION THREAD

**Discussion by mnd5756242** [projectdiscovery/nuclei#5235] [Q&A]
The following is what I call after executing， the result is nil 

In other words, how do I modify the code to call the callback function and obtain the result after the reverse connection?

`func (r *Runner) ExecuteNucleiPoc(target string, poc *nuclei.Template) (results []*output.ResultEvent, err error) {
	e := poc.Executer
	ctx := scan.NewScanContext(contextargs.NewWithInput(target))

	ctx.OnResult = func(event *output.InternalWrappedEvent) {
	            if resultValue, ok := event.InternalEvent["result"]; ok {
			if result, ok := resultValue.(*output.ResultEvent); ok {
				results = append(results, result)
			}
		}
	}
	results, err = e.ExecuteWithResults(ctx)
	if err != nil {
		return nil, err
	}

	return results, err
}`

AND I initialized interactClient as follows：
`//outputWriter := testutils.NewMockOutputWriter(options.OmitTemplate)
	outputWriter, errs := output.NewStandardWriter(options)
	if errs != nil {
		logger.Error(err, "could not create output file")
	}

	_ = protocolstate.Init(options)
	_ = protocolinit.Init(options)

	interactOpts := interactsh.DefaultOptions(outputWriter, reportingClient, mockProgress)
	interactClient, err := interactsh.New(interactOpts)
	if err != nil {
		return
	}
	defer interactClient.Close()

	catalog := disk.NewCatalog("")

	ExecuterOptions = protocols.ExecutorOptions{
		Output:       outputWriter,
		Options:      options,
		Progress:     mockProgress,
		Catalog:      catalog,
		IssuesClient: reportingClient,
		Interactsh:   interactClient,
		RateLimiter:  ratelimit.New(context.Background(), uint(o.Thread), time.Second),
		//HostErrorsCache: cache,
		Colorizer: aurora.NewAurora(true),
		ResumeCfg: types.NewResumeCfg(),
	}`
So is there no callback result because Authorization is not configured？I use nuclei reverse.

**Comment by ehsandeep** [comment]
@mnd5756242 please use https://github.com/projectdiscovery/nuclei/discussions/categories/q-a for questions instead of github issues.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
